File: ruby/lib/catalogue_actions/catalogue_actions.rb
require_relative './display_product_details_command'
require_relative './list_products_command'

module CatalogueActions

	class CatalogueActions

		def initialize(catalogue)
			@catalogue = catalogue
		end

		def display_product_details
			DisplayProductDetailsCommand.new(@catalogue)
		end

		def list_products
			ListProductsCommand.new(@catalogue)
		end

	end
end

File: python3/catalogue_actions/catalogue_actions.py
from . display_product_details_command import DisplayProductDetailsCommand
from . list_products_command import ListProductsCommand

class CatalogueActions:

    def __init__(self, catalogue):
        self.catalogue = catalogue

    def displayProductDetails(self):
        return DisplayProductDetailsCommand(self.catalogue)

    def listProducts(self):
        return ListProductsCommand(self.catalogue)

